docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
-    positionals: [
-        {
-            name: "oldDataSet",
-            description: "The old data set",
-            type: "string"
-        },
-        {
-            name: "newDataSet",
-            description: "The new data set",
-            type: "string"
+            // convert the old data set name to use as a path/file
+            let file = oldDataSet.replace(regex, "/");
+            file = file.replace(regex2, ".") + "dat";
+            // Load the downloaded contents of 'oldDataSet'
+            const oldContent = readFileSync(`${file}`).toString();
+
+            // convert the new data set name to use as a path/file
+            file = newDataSet.replace(regex, "/");
+            file = file.replace(regex2, ".") + "dat";
+            // Load the downloaded contents of 'oldDataSet'
+            const newContent = readFileSync(`${file}`).toString();
+
+            return diff.createTwoFilesPatch(oldDataSet, newDataSet, oldContent, newContent, "Old", "New");
         }
-    ],
-    profile: {
-        required: ["zosmf"]
     }
-};
-```
+    ```
 
-`DataSets.handler.ts` should contain the following:
-```typescript
-import { ICommandHandler, IHandlerParameters, TextUtils, Session } from "@zowe/imperative";
-import { DataSetDiff } from "../../../api/DataSetDiff";
-
-export default class DataSetsDiffHandler implements ICommandHandler {
-    public async process(params: IHandlerParameters): Promise<void> {
-
-        const profile = params.profiles.get("zosmf");
-        const session = new Session({
-            type: "basic",
-            hostname: profile.host,
-            port: profile.port,
-            user: profile.user,
-            password: profile.pass,
-            base64EncodedAuth: profile.auth,
-            rejectUnauthorized: profile.rejectUnauthorized,
-        });
-        const resp = await DataSetDiff.diff(session, params.arguments.oldDataSet, params.arguments.newDataSet);
-        params.response.console.log(resp);
-    }
-}
+3. In the `src` directory, add the following code to the end of the `index.ts` file to make the API available for other developers to import:
 
-```
+    ```typescript
+    export * from "./api/DataSetDiff";
+    ```
+
+    When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
+
+**To create a unique plug-in:** The file name and code in Step 2 may be entirely different if you want to implement an API to do something else.
+
+## Building your plug-in source
+
+Confirm that your project builds successfully.
+
+Follow these steps:
+
+1. In the terminal, run `npm run build` to verify there are no errors.
+
+    This command builds your typescript files into javascript files for your plug-in in the `lib` directory.
+
+2. If you come across linting errors, run `npm run lint -- --fix` to resolve the errors automatically.
+
+    When successful, no errors are returned and the `lib` directory contains the built javascript files.
+
+**To create a unique plug-in:** Follow these same steps.
```

Comment:
I guess we want to keep a `to create a unique plug-in` for consistency. And also because we said we will have that note after each section (in the note at the bottom of the `Overview` section).
`Note: If you are ready to create your own unique Zowe CLI plug-in, refer to the notes below each tutorial step for guidance.`